db/crud.go

```diff
@@ -910,19 +907,11 @@ func (db *Database) PutExistingRevWithBody(docid string, body Body, docHistory [
 
 	doc, newRevID, putExistingRevErr := db.PutExistingRev(newDoc, docHistory, noConflicts)
 
-	// Callers expect the below properties to be in the body even if PutExistingRev returns error
-	body[BodyId] = docid
-	body[BodyRev] = newRevID
-
-	if deleted {
-		body[BodyDeleted] = deleted
-	}
-
 	if putExistingRevErr != nil {
-		return nil, putExistingRevErr
+		return nil, newRevID, putExistingRevErr
```

Comment:
Let's return an empty string here just to be a bit safer.

```suggestion
		return nil, "", putExistingRevErr
```